src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/ConnectionHandlerContext.cs

```diff
@@ -10,7 +10,7 @@ namespace Microsoft.Data.SqlClientX.Handlers
 {    /// <summary>
      /// Class that contains all context data needed for various handlers.
      /// </summary>
-    internal class ConnectionHandlerContext : HandlerRequest
+    internal class ConnectionHandlerContext : HandlerRequest, ICloneable
```

Comment:
The use of a default constructor in the ConnectionHandlerContext class allows the creation of an instance where all properties are null. This implies that an object with all its properties set to null is considered valid, which is likely not the case. Allowing such states can lead to runtime errors and logical inconsistencies within the application.

Reply:
I think this is expected for the handler context. It is going to be a mostly data-only object that is passed between handlers, so having all properties set to null or default is likely a valid initial state. Additionally allowing all properties to be set without using a non-default constructor, allows for easier testing.

Reply:
I really doubt that null object is nulll because it is not able to do anything. Unless you think context is just a dictionary and does not have any logic. 
Context should ensure that it never gets in an invalid state and hence I forsee it has business logic otherwise there is a risk of duplicating the business across different handlers.

Reply:
At least it needs to have a Connection String.

Reply:
I added a to-do for it. Want to make sure how Orchestrator/chain needs to be initialized by the SqlConnector, will add necessary constructors with it.